packages/devtools_app/lib/src/shared/common_widgets.dart

```diff
@@ -239,6 +239,33 @@ class ToolbarRefresh extends ToolbarAction {
   });
 }
 
+class StartStopRecordingButton extends GaDevToolsButton {
+  StartStopRecordingButton({
+    super.key,
+    required this.recording,
+    required super.onPressed,
+    required super.gaScreen,
+    required super.gaSelection,
+    super.minScreenWidthForTextBeforeScaling,
+    String? tooltipOverride,
+    Color? colorOverride,
+    String? labelOverride,
+  }) : super(
+          icon: _icon(recording),
+          label: labelOverride ?? _label(recording),
+          color: colorOverride ?? _color(recording),
+          tooltip: tooltipOverride ?? _tooltip(recording),
+        );
+  final bool recording;
+  static IconData _icon(bool recording) =>
+      recording ? Icons.stop : Icons.fiber_manual_record;
+  static String _label(bool recording) =>
+      recording ? 'Stop recording' : 'Start recording';
+  static String _tooltip(bool recording) =>
+      recording ? 'Stop recording' : 'Start recording';
+  static Color? _color(bool recording) => recording ? Colors.red : null;
```

Comment:
style nit: class member ordering: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#order-other-class-members-in-a-way-that-makes-sense

- constructors
- static methods
- final fields

also, prefer new lines between different types of class members.
```
constructor
<new line>
static methods
<new line>
final fields
```